pkg/pillar/pubsub/pubsub.go

```diff
@@ -431,6 +441,16 @@ func subscribeImpl(agentName string, agentScope string, topicType interface{},
 	sub.userCtx = ctx
 	sub.km = keyMap{key: NewLockedStringMap()}
 	sub.persistent = persistent
+	if options != nil {
+		sub.CreateHandler = options.CreateHandler
+		sub.ModifyHandler = options.ModifyHandler
+		sub.DeleteHandler = options.DeleteHandler
+		sub.RestartHandler = options.RestartHandler
+		sub.SynchronizedHandler = options.SyncHandler
+		sub.MaxProcessTimeWarn = options.WarningTime
+		sub.MaxProcessTimeError = options.ErrorTime
```

Comment:
Is this a good place to check options.WarningTime and options.ErrorTime are not set ( set to 0 ) and if yes - just set them to defaults ?? That way, most places, we don't need to set these at all..

Reply:
It probably is. I was going to do it, but didn't want to change everything else now, mixing too much in one PR. But I think it makes sense. Will be done in 5.

Reply:
I take it back. Let's not do it now. The problem is that there is no `nil` value for `time.Duration`, which means we have no way of knowing if the caller meant to pass `0` or nothing. We need to figure out what the correct default should be, and how we distinguish between "no timeout" and "default timeout".

Let's make it a separate PR focused on that.